src/memory.rs

```diff
@@ -524,44 +538,44 @@ const BLOCK_SIZE: usize = 64;
 #[derive(Clone, Debug)]
 pub struct UndefMask {
```

Comment:
In the `EvalContext` some things got simpler. But in the `Memory` I agree, all of the things became harder to follow.

I started changing everything to `Size` where it felt consistent, but then that required a lot of conversions everywhere. I can undo it and convert whenever we access the layout instead.